### PR TITLE
M3 num dist by cat

### PR DESCRIPTION
**Pull request by morrismanfung**

### CODE REVIEW

**Comment by MNBhat** [comment]
The errors encountered when performing check on the function  
`error message :  UseMethod("mutate") :`
`Error in library(car): there is no package called 'car'`

File: R/num_dist_by_cat.R [hunk 1]
```diff
@@ -1,26 +1,94 @@
...
+#' @param group Name of the column for the categorical variable.
 #' @param data Target data frame for visualization.
 #' @param title_hist Title for the histogram.
 #' @param title_boxplot Title for the boxplot.
 #' @param lab_num Axis label for the numeric variable.
-#' @param lab_cat Axis label for the categorical variable.
+#' @param lab_group Axis label for the categorical variable.
 #' @param num_on_x Whether the numeric variable is put on the x-axis in the boxplot.
 #' @param stat Whether printing the test statistic and summary or not.
 #'
```
Comment (austin-shih) [inline R/num_dist_by_cat.R]: remove empty line

File: R/num_dist_by_cat.R [hunk 2]
```diff
@@ -1,26 +1,94 @@
...
 #' @return A ggplot object consists of a histogram and a boxplot.
 #' @export
```
Comment (austin-shih) [inline R/num_dist_by_cat.R:12]: move @export above @return

File: R/num_dist_by_cat.R [hunk 3]
```diff
@@ -1,26 +1,94 @@
...
 #'
```
Comment (austin-shih) [inline R/num_dist_by_cat.R:14]: remove empty line

File: R/num_dist_by_cat.R [hunk 4]
```diff
@@ -1,26 +1,94 @@
...
 #' @examples
```
Comment (austin-shih) [inline R/num_dist_by_cat.R:15]: add one more space in front of @example. EXACTLY one space so it's indented compared to all the other lines

**Comment by morrismanfung** [comment]
> The errors encountered when performing check on the function `error message : UseMethod("mutate") :` `Error in library(car): there is no package called 'car'`

I checked the code in a new environment and figured out the problem is that `cowplot` and `car` are not installed in the new environment. I am not sure whether they will be automatically installed when someone actually install our pacakge. `load_all()` will not install any function I think. Instead, we need to run `install_deps` manually.

Reference:
https://github.com/r-lib/devtools/issues/1380
https://rdrr.io/cran/devtools/man/install_deps.html

**Comment by morrismanfung** [comment]
Try to run `install()` before `load_all()` and `check()`. I think it will better mimic the actual installation process. Thanks @austin-shih for the suggestion.

**Review by austin-shih — APPROVED** [review]
looks good

**Pull request merged**